support/proxy/proxy.go

```diff
@@ -0,0 +1,63 @@
+package proxy
+
+import (
+	"os"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func SetEnvVars(env *[]corev1.EnvVar, additionalNoProxy ...string) {
+	SetEnvVarsTo(env,
+		os.Getenv("HTTP_PROXY"),
+		os.Getenv("HTTPS_PROXY"),
+		os.Getenv("NO_PROXY")+","+strings.Join(append(additionalNoProxy, "kube-apiserver"), ","),
+	)
+}
+
+func SetEnvVarsTo(env *[]corev1.EnvVar, httpProxy, httpsProxy, noProxy string) {
+	if httpProxy == "" {
+		removeEnvVarIfPresent(env, "HTTP_PROXY")
```

Comment:
would just update to "" avoid the need to remove?

Reply:
That will cause the hypershift operator to always update itself once and I wasn't sure if there might be code somewhere that differentiates between empty and unset